code/datums/elements/footstep.dm

```diff
@@ -134,21 +134,31 @@
 	if(!source_loc)
 		return
 
-	play_fov_effect(source, 5, "footstep", direction, ignore_self = TRUE)
+	//cache for sanic speed (lists are references anyways)
+	var/static/list/footstep_sounds = GLOB.footstep
+	///list returned by playsound() filled by client mobs who heard the footstep. given to play_fov_effect()
+	var/list/heard_clients
+
 	if ((source.wear_suit?.body_parts_covered | source.w_uniform?.body_parts_covered | source.shoes?.body_parts_covered) & FEET)
 		// we are wearing shoes
-		playsound(source_loc, pick(GLOB.footstep[source_loc.footstep][1]),
-			GLOB.footstep[source_loc.footstep][2] * volume * volume_multiplier,
+
+		heard_clients = playsound(source_loc, pick(footstep_sounds[source_loc.footstep][1]),
+			footstep_sounds[source_loc.footstep][2] * volume * volume_multiplier,
 			TRUE,
-			GLOB.footstep[source_loc.footstep][3] + e_range + range_adjustment, falloff_distance = 1, vary = sound_vary)
+			footstep_sounds[source_loc.footstep][3] + e_range + range_adjustment, falloff_distance = 1, vary = sound_vary)
 	else
 		if(source.dna.species.special_step_sounds)
-			playsound(source_loc, pick(source.dna.species.special_step_sounds), 50, TRUE, falloff_distance = 1, vary = sound_vary)
+			heard_clients = playsound(source_loc, pick(source.dna.species.special_step_sounds), 50, TRUE, falloff_distance = 1, vary = sound_vary)
 		else
-			playsound(source_loc, pick(GLOB.barefootstep[source_loc.barefootstep][1]),
-				GLOB.barefootstep[source_loc.barefootstep][2] * volume * volume_multiplier,
+			var/static/list/bare_footstep_sounds = GLOB.barefootstep
+
+			heard_clients = playsound(source_loc, pick(bare_footstep_sounds[source_loc.barefootstep][1]),
+				bare_footstep_sounds[source_loc.barefootstep][2] * volume * volume_multiplier,
 				TRUE,
-				GLOB.barefootstep[source_loc.barefootstep][3] + e_range + range_adjustment, falloff_distance = 1, vary = sound_vary)
+				bare_footstep_sounds[source_loc.barefootstep][3] + e_range + range_adjustment, falloff_distance = 1, vary = sound_vary)
+
+	if(heard_clients)
+		play_fov_effect(source, 5, "footstep", direction, ignore_self = TRUE, override_list = heard_clients)
```

Comment:
playsound() now returns the actual listeners that head the message. now whenever we footstep we dont have to call get_hearers_in_view() to find fov see-ers

Reply:
chad